src/ISDisplay.cpp

```diff
@@ -653,7 +657,8 @@ string cInertialSenseDisplay::DataToString(const p_data_t* data)
             str = DataToStringRawHex((const char *)data->ptr, data->hdr, 32);
 		else if (m_editData.did == data->hdr.id)	
 		{	// Default view
-			str = DatasetToString(m_editData.pData = data);
+			copyDataPToDataP(&m_editData.pData, data, MAX_DATASET_SIZE);
```

Comment:
We discussed this, but I didn't see the code above on line 477.  Why are we doing the call to copyDataPToDataP() twice?  Can we modify 477 to be true, if either m_editData.did OR m_outputOnceDid matches the hdr.id?  Or does that not jive with the PRETTY_PRINT on line 526?   It seems like in EVERY CASE, we could/should copy the data into m_editData.pData, and then display based on that.

Reply:
copyDataPToDataP() is new helper function because we've never copied from p_data_t to p_data_t.

You are correct in that the second instance of copyDataPToDataP() was extra and not needed.  I removed the extra data copy and retested the different display modes.